parlai/core/teachers.py

```diff
@@ -489,11 +489,11 @@ def next_example(self):
             and self._episode_done
             and self.episode_idx + self.opt.get("batchsize", 1) >= self.num_episodes()
         ):
-            epoch_done = True
+            self.epochDone = True
```

Comment:
mmm i'm not sure about this

Reply:
Currently `epochDone` isn't updated to `True` anywhere in `FixedDialogTeacher`. Do you know what's up with that? Why would it _partially_ manage its own private state?

Reply:
what exactly is this change for?

Reply:
So that `iter(Teacher)` stops when it should stop and loops forever when it should loop forever.

It also seems like it's what _should_ happen. Otherwise `FixedDialogTeacher` relies on `get_orig_action` (in `act`) to set this properly. If a subclass overrides `act` but doesn't override `next_example`, it will need to manage `epochDone` itself.

But let me know if there's something I'm missing!

Reply:
I'm still really nervous about this 😅 teachers is a bit of a mess and I'd like to move us towards deprecating all FDTs from the code...